lib/compress/zstd_compress.c

```diff
@@ -2372,9 +2381,21 @@ static size_t ZSTD_compress_frameChunk (ZSTD_CCtx* cctx,
                 cSize = ZSTD_noCompressBlock(op, dstCapacity, ip, blockSize, lastBlock);
                 FORWARD_IF_ERROR(cSize);
             } else {
-                U32 const cBlockHeader24 = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
-                MEM_writeLE24(op, cBlockHeader24);
-                cSize += ZSTD_blockHeaderSize;
+                U32 cBlockHeader;
+                /*
+                    We check to see if the regularly compressed block size is
+                    less than 10 (the upper bound for RLE blocks) and we check
+                    to see if the block is an RLE
+                */
+                if (cSize < 10 && ZSTD_isRLE(ip, blockSize)) {
+                    op[ZSTD_blockHeaderSize] = ip[0];
+                    cBlockHeader = lastBlock + (((U32)bt_rle)<<1) + (U32)(blockSize << 3);
+                    cSize = ZSTD_blockHeaderSize + 1;
+                } else {
+                    cBlockHeader = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
```

Comment:
Add `assert(!ZSTD_isRLE(ip, blockSize))` here so we can know if the upper bound is wrong.